File: app/src/main/java/lazy/of/go/to/module/login/LoginModule.java
package lazy.of.go.to.module.login;

import dagger.Module;
import dagger.Provides;
import lazy.of.go.to.app.login.LoginActivity;
import lazy.of.go.to.app.login.LoginView;
import lazy.of.go.to.auth.LazyAuth;
import lazy.of.go.to.auth.firebase.FbAuth;
import lazy.of.go.to.common.Log;
import lazy.of.go.to.di.ActivityScoped;

/**
 * @author piece.of.lazy
 */
@Module
public abstract class LoginModule {

    @ActivityScoped
    @Provides
    static LazyAuth provideAuth() {
        return new FbAuth();
    }
}

File: app/src/main/kotlin/lazy/of/go/to/app/login/LoginModule.kt
package lazy.of.go.to.app.login

import dagger.Module
import dagger.Provides
import lazy.of.go.to.db.DbInjection
import lazy.of.go.to.di.ActivityScoped
import lazy.of.go.to.domain.data.DbUser

/**
 * @author piece.of.lazy
 */
@Module
abstract class LoginModule {
    @Module
    companion object {
        @JvmStatic
        @ActivityScoped
        @Provides
        internal fun provideDbUser(dbInjection: DbInjection) = dbInjection.getDB(DbUser::class)
    }
}